src/index.tsx

```diff
@@ -34,7 +34,10 @@ function returnEmptyString() {
   return '';
 }
 
-function returnDocument() {
+function returnDocument(element?: HTMLElement) {
+  if (element) {
+    return element.ownerDocument;
```

Comment:
should be 
```suggestion
    return element.getRootNode?.() || element.ownerDocument;
```

Reply:
ref https://github.com/ant-design/ant-design/issues/28934#issuecomment-767390087

Reply:
@JohnDeved

popup can not be created in shadow dom.

the reason we need this package for popup is because sibling element can be blocked because of parent's scroll area, or other element above, so we need something to make sure the popup is on the top.

showing popup in shadow dom means it won't be on top.